smarts/core/agent_manager.py

```diff
@@ -348,15 +351,46 @@ def _filter_social_agent_actions_for_controlled_vehicles(
 
         return social_agent_actions
 
-    def send_observations_to_social_agents(self, observations):
-        """Forwards observations to managed social agents."""
-        # TODO: Don't send observations (or receive actions) from agents that have done
-        #       vehicles.
+    def add_social_observation_callback(self, callback, id):
+        """Suscribe to observe social agents."""
+        self._social_agent_observation_callbacks[id] = callback
+
+    def remove_social_observation_callback(self, id):
+        """Remove a subscription to social agents."""
+        del self._social_agent_observation_callbacks[id]
```

Comment:
```suggestion
    def remove_social_observation_callback(self, agent_id: str):
        """Remove a subscription to social agents."""
        del self._social_agent_observation_callbacks[agent_id]
```

Reply:
I realise id is misleading and should be `callback_id`